File: wf-api-integration/references/statement-management/inquiry-statement-detail/java/model/response/InquiryStatementDetailResponse.java
package {basePackage}.wf.model.response;

import org.apache.commons.lang3.builder.ToStringBuilder;
import org.apache.commons.lang3.builder.ToStringStyle;
import {basePackage}.wf.model.domain.Amount;
import {basePackage}.wf.model.domain.ForeignExchangeQuote;
import {basePackage}.wf.model.domain.FundMoveDetail;
import {basePackage}.wf.model.domain.OperatorInfo;
import {basePackage}.wf.model.domain.RelatedStatement;

import java.util.List;

/**
 * WorldFirst inquiryStatementDetail 响应对象
 *
 */
public class InquiryStatementDetailResponse {

    /** 接口调用结果 */
    private Result result;

    /** 响应唯一 ID，最大 32 位 */
    private String responseId;

    /**
     * 交易 ID，TRANSFER/TRANSFER_REFUND/WITHDRAWAL/WITHDRAWAL_REFUND/
     * CONVERSION/CONVERSION_DEAL/CHARGE/CHARGE_REFUND/DEDUCTION/FUND_COLLECTION 类型必返回
     */
    private String transactionId;

    /** 调用方传入的外部交易 ID，最大 256 位 */
    private String extTransactionId;

    /** 交易状态：INIT / PROCESSING / PENDING / SUCCESS / FAIL / REFUNDED */
    private String transactionStatus;

    /** 余额变动时间，ISO 8601 格式，如 2019-11-27T12:01:01+08:00 */
    private String transactionTime;

    /** 交易类型，如 TRANSFER、COLLECTION 等 */
    private String transactionType;

    /** 本次余额变动金额 */
    private Amount transactionAmount;

    /** 原始提交的交易金额 */
    private Amount originalTransactionAmount;

    /**
     * 手续费金额，TRANSFER/TRANSFER_REFUND/WITHDRAWAL/WITHDRAWAL_REFUND/
     * CONVERSION/CONVERSION_DEAL/CHARGE/CHARGE_REFUND/DEDUCTION/FUND_COLLECTION 类型条件必返回
     */
    private Amount feeAmount;

    /** 手续费类型：OBO_SERVICE_FEE / REMIT_SERVICE_FEE */
    private String feeItemType;

    /** 扣除手续费后的净额，TRANSFER/CHARGE/PAYMENT/CASH_BACK 类型必返回 */
    private Amount netAmount;

    /** 换汇后实际到账金额，TRANSFER/CHARGE/PAYMENT/CASH_BACK 类型必返回 */
    private Amount receiveAmount;

    /** 交易后账户实时余额 */
    private Amount accountBalance;

    /** 资金流动详情（付款方/收款方信息） */
    private FundMoveDetail fundMoveDetail;

    /** 汇率报价信息（TRANSFER/WITHDRAWAL/CONVERSION/CONVERSION_DEAL/CHARGE 类型可选） */
    private ForeignExchangeQuote foreignExchangeQuote;

    /** 退款汇率报价（TRANSFER_REFUND/WITHDRAWAL_REFUND/CHARGE_REFUND 类型可选） */
    private ForeignExchangeQuote refundForeignExchangeQuote;

    /**
     * 余额类型：NORMAL_BALANCE（默认）/ SAME_NAME_TOP_UP_BALANCE / BUDGET_BALANCE
     */
    private String balanceType;

    /** 账单流水唯一 ID */
    private String accountingBizNo;

    /** 失败原因，transactionStatus=FAIL 时返回 */
    private Result failReason;

    /** 关联交易列表 */
    private List<RelatedStatement> combinedTransactionList;

    /** 操作员信息，仅通过 WF 门户操作时返回 */
    private OperatorInfo operatorInfo;

    /** 商品名称 */
    private String goodsName;

    /** 商品金额 */
    private Amount goodsAmount;

    /** 优惠前手续费金额 */
    private Amount originalFeeAmount;

    /** 手续费优惠金额 */
    private Amount discountFeeAmount;

    /**
     * Getter method for property <tt>result</tt>.
     *
     * @return property value of result
     */
    public Result getResult() {
        return result;
    }

    /**
     * Setter method for property <tt>result</tt>.
     *
     * @param result value to be assigned to property result
     */
    public void setResult(Result result) {
        this.result = result;
    }

    /**
     * Getter method for property <tt>responseId</tt>.
     *
     * @return property value of responseId
     */
    public String getResponseId() {
        return responseId;
    }

    /**
     * Setter method for property <tt>responseId</tt>.
     *
     * @param responseId value to be assigned to property responseId
     */
    public void setResponseId(String responseId) {
        this.responseId = responseId;
    }

    /**
     * Getter method for property <tt>transactionId</tt>.
     *
     * @return property value of transactionId
     */
    public String getTransactionId() {
        return transactionId;
    }

    /**
     * Setter method for property <tt>transactionId</tt>.
     *
     * @param transactionId value to be assigned to property transactionId
     */
    public void setTransactionId(String transactionId) {
        this.transactionId = transactionId;
    }

    /**
     * Getter method for property <tt>extTransactionId</tt>.
     *
     * @return property value of extTransactionId
     */
    public String getExtTransactionId() {
        return extTransactionId;
    }

    /**
     * Setter method for property <tt>extTransactionId</tt>.
     *
     * @param extTransactionId value to be assigned to property extTransactionId
     */
    public void setExtTransactionId(String extTransactionId) {
        this.extTransactionId = extTransactionId;
    }

    /**
     * Getter method for property <tt>transactionStatus</tt>.
     *
     * @return property value of transactionStatus
     */
    public String getTransactionStatus() {
        return transactionStatus;
    }

    /**
     * Setter method for property <tt>transactionStatus</tt>.
     *
     * @param transactionStatus value to be assigned to property transactionStatus
     */
    public void setTransactionStatus(String transactionStatus) {
        this.transactionStatus = transactionStatus;
    }

    /**
     * Getter method for property <tt>transactionTime</tt>.
     *
     * @return property value of transactionTime
     */
    public String getTransactionTime() {
        return transactionTime;
    }

    /**
     * Setter method for property <tt>transactionTime</tt>.
     *
     * @param transactionTime value to be assigned to property transactionTime
     */
    public void setTransactionTime(String transactionTime) {
        this.transactionTime = transactionTime;
    }

    /**
     * Getter method for property <tt>transactionType</tt>.
     *
     * @return property value of transactionType
     */
    public String getTransactionType() {
        return transactionType;
    }

    /**
     * Setter method for property <tt>transactionType</tt>.
     *
     * @param transactionType value to be assigned to property transactionType
     */
    public void setTransactionType(String transactionType) {
        this.transactionType = transactionType;
    }

    /**
     * Getter method for property <tt>transactionAmount</tt>.
     *
     * @return property value of transactionAmount
     */
    public Amount getTransactionAmount() {
        return transactionAmount;
    }

    /**
     * Setter method for property <tt>transactionAmount</tt>.
     *
     * @param transactionAmount value to be assigned to property transactionAmount
     */
    public void setTransactionAmount(Amount transactionAmount) {
        this.transactionAmount = transactionAmount;
    }

    /**
     * Getter method for property <tt>originalTransactionAmount</tt>.
     *
     * @return property value of originalTransactionAmount
     */
    public Amount getOriginalTransactionAmount() {
        return originalTransactionAmount;
    }

    /**
     * Setter method for property <tt>originalTransactionAmount</tt>.
     *
     * @param originalTransactionAmount value to be assigned to property originalTransactionAmount
     */
    public void setOriginalTransactionAmount(Amount originalTransactionAmount) {
        this.originalTransactionAmount = originalTransactionAmount;
    }

    /**
     * Getter method for property <tt>feeAmount</tt>.
     *
     * @return property value of feeAmount
     */
    public Amount getFeeAmount() {
        return feeAmount;
    }

    /**
     * Setter method for property <tt>feeAmount</tt>.
     *
     * @param feeAmount value to be assigned to property feeAmount
     */
    public void setFeeAmount(Amount feeAmount) {
        this.feeAmount = feeAmount;
    }

    /**
     * Getter method for property <tt>feeItemType</tt>.
     *
     * @return property value of feeItemType
     */
    public String getFeeItemType() {
        return feeItemType;
    }

    /**
     * Setter method for property <tt>feeItemType</tt>.
     *
     * @param feeItemType value to be assigned to property feeItemType
     */
    public void setFeeItemType(String feeItemType) {
        this.feeItemType = feeItemType;
    }

    /**
     * Getter method for property <tt>netAmount</tt>.
     *
     * @return property value of netAmount
     */
    public Amount getNetAmount() {
        return netAmount;
    }

    /**
     * Setter method for property <tt>netAmount</tt>.
     *
     * @param netAmount value to be assigned to property netAmount
     */
    public void setNetAmount(Amount netAmount) {
        this.netAmount = netAmount;
    }

    /**
     * Getter method for property <tt>receiveAmount</tt>.
     *
     * @return property value of receiveAmount
     */
    public Amount getReceiveAmount() {
        return receiveAmount;
    }

    /**
     * Setter method for property <tt>receiveAmount</tt>.
     *
     * @param receiveAmount value to be assigned to property receiveAmount
     */
    public void setReceiveAmount(Amount receiveAmount) {
        this.receiveAmount = receiveAmount;
    }

    /**
     * Getter method for property <tt>accountBalance</tt>.
     *
     * @return property value of accountBalance
     */
    public Amount getAccountBalance() {
        return accountBalance;
    }

    /**
     * Setter method for property <tt>accountBalance</tt>.
     *
     * @param accountBalance value to be assigned to property accountBalance
     */
    public void setAccountBalance(Amount accountBalance) {
        this.accountBalance = accountBalance;
    }

    /**
     * Getter method for property <tt>fundMoveDetail</tt>.
     *
     * @return property value of fundMoveDetail
     */
    public FundMoveDetail getFundMoveDetail() {
        return fundMoveDetail;
    }

    /**
     * Setter method for property <tt>fundMoveDetail</tt>.
     *
     * @param fundMoveDetail value to be assigned to property fundMoveDetail
     */
    public void setFundMoveDetail(FundMoveDetail fundMoveDetail) {
        this.fundMoveDetail = fundMoveDetail;
    }

    /**
     * Getter method for property <tt>foreignExchangeQuote</tt>.
     *
     * @return property value of foreignExchangeQuote
     */
    public ForeignExchangeQuote getForeignExchangeQuote() {
        return foreignExchangeQuote;
    }

    /**
     * Setter method for property <tt>foreignExchangeQuote</tt>.
     *
     * @param foreignExchangeQuote value to be assigned to property foreignExchangeQuote
     */
    public void setForeignExchangeQuote(ForeignExchangeQuote foreignExchangeQuote) {
        this.foreignExchangeQuote = foreignExchangeQuote;
    }

    /**
     * Getter method for property <tt>refundForeignExchangeQuote</tt>.
     *
     * @return property value of refundForeignExchangeQuote
     */
    public ForeignExchangeQuote getRefundForeignExchangeQuote() {
        return refundForeignExchangeQuote;
    }

    /**
     * Setter method for property <tt>refundForeignExchangeQuote</tt>.
     *
     * @param refundForeignExchangeQuote value to be assigned to property refundForeignExchangeQuote
     */
    public void setRefundForeignExchangeQuote(ForeignExchangeQuote refundForeignExchangeQuote) {
        this.refundForeignExchangeQuote = refundForeignExchangeQuote;
    }

    /**
     * Getter method for property <tt>balanceType</tt>.
     *
     * @return property value of balanceType
     */
    public String getBalanceType() {
        return balanceType;
    }

    /**
     * Setter method for property <tt>balanceType</tt>.
     *
     * @param balanceType value to be assigned to property balanceType
     */
    public void setBalanceType(String balanceType) {
        this.balanceType = balanceType;
    }

    /**
     * Getter method for property <tt>accountingBizNo</tt>.
     *
     * @return property value of accountingBizNo
     */
    public String getAccountingBizNo() {
        return accountingBizNo;
    }

    /**
     * Setter method for property <tt>accountingBizNo</tt>.
     *
     * @param accountingBizNo value to be assigned to property accountingBizNo
     */
    public void setAccountingBizNo(String accountingBizNo) {
        this.accountingBizNo = accountingBizNo;
    }

    /**
     * Getter method for property <tt>failReason</tt>.
     *
     * @return property value of failReason
     */
    public Result getFailReason() {
        return failReason;
    }

    /**
     * Setter method for property <tt>failReason</tt>.
     *
     * @param failReason value to be assigned to property failReason
     */
    public void setFailReason(Result failReason) {
        this.failReason = failReason;
    }

    /**
     * Getter method for property <tt>combinedTransactionList</tt>.
     *
     * @return property value of combinedTransactionList
     */
    public List<RelatedStatement> getCombinedTransactionList() {
        return combinedTransactionList;
    }

    /**
     * Setter method for property <tt>combinedTransactionList</tt>.
     *
     * @param combinedTransactionList value to be assigned to property combinedTransactionList
     */
    public void setCombinedTransactionList(List<RelatedStatement> combinedTransactionList) {
        this.combinedTransactionList = combinedTransactionList;
    }

    /**
     * Getter method for property <tt>operatorInfo</tt>.
     *
     * @return property value of operatorInfo
     */
    public OperatorInfo getOperatorInfo() {
        return operatorInfo;
    }

    /**
     * Setter method for property <tt>operatorInfo</tt>.
     *
     * @param operatorInfo value to be assigned to property operatorInfo
     */
    public void setOperatorInfo(OperatorInfo operatorInfo) {
        this.operatorInfo = operatorInfo;
    }

    /**
     * Getter method for property <tt>goodsName</tt>.
     *
     * @return property value of goodsName
     */
    public String getGoodsName() {
        return goodsName;
    }

    /**
     * Setter method for property <tt>goodsName</tt>.
     *
     * @param goodsName value to be assigned to property goodsName
     */
    public void setGoodsName(String goodsName) {
        this.goodsName = goodsName;
    }

    /**
     * Getter method for property <tt>goodsAmount</tt>.
     *
     * @return property value of goodsAmount
     */
    public Amount getGoodsAmount() {
        return goodsAmount;
    }

    /**
     * Setter method for property <tt>goodsAmount</tt>.
     *
     * @param goodsAmount value to be assigned to property goodsAmount
     */
    public void setGoodsAmount(Amount goodsAmount) {
        this.goodsAmount = goodsAmount;
    }

    /**
     * Getter method for property <tt>originalFeeAmount</tt>.
     *
     * @return property value of originalFeeAmount
     */
    public Amount getOriginalFeeAmount() {
        return originalFeeAmount;
    }

    /**
     * Setter method for property <tt>originalFeeAmount</tt>.
     *
     * @param originalFeeAmount value to be assigned to property originalFeeAmount
     */
    public void setOriginalFeeAmount(Amount originalFeeAmount) {
        this.originalFeeAmount = originalFeeAmount;
    }

    /**
     * Getter method for property <tt>discountFeeAmount</tt>.
     *
     * @return property value of discountFeeAmount
     */
    public Amount getDiscountFeeAmount() {
        return discountFeeAmount;
    }

    /**
     * Setter method for property <tt>discountFeeAmount</tt>.
     *
     * @param discountFeeAmount value to be assigned to property discountFeeAmount
     */
    public void setDiscountFeeAmount(Amount discountFeeAmount) {
        this.discountFeeAmount = discountFeeAmount;
    }

    @Override
    public String toString() {
        return ToStringBuilder.reflectionToString(this, ToStringStyle.SHORT_PREFIX_STYLE);
    }
}

File: wf-api-integration/references/statement-management/inquiry-statement-detail/python/model/response/inquiry_statement_detail_response.py
# -*- coding: utf-8 -*-
"""
WorldFirst inquiryStatementDetail 响应模型
"""

from dataclasses import dataclass, field
from typing import List, Optional

from {basePackage}.wf.model.response.result import Result
from {basePackage}.wf.model.domain.statement_record import (
    Amount,
    ForeignExchangeQuote,
    FundMoveDetail,
    OperatorInfo,
)
from {basePackage}.wf.model.domain.related_statement import RelatedStatement


@dataclass
class InquiryStatementDetailResponse:
    """inquiryStatementDetail 响应

    Attributes:
        result: 通用响应结果
        response_id: 响应唯一 ID，最大 32 位
        transaction_id: 交易 ID
        ext_transaction_id: 外部交易 ID，最大 256 位
        transaction_status: 交易状态（INIT/PROCESSING/PENDING/SUCCESS/FAIL/REFUNDED）
        transaction_time: 余额变动时间（ISO 8601）
        transaction_type: 交易类型（TRANSFER, COLLECTION 等）
        transaction_amount: 本次余额变动金额
        original_transaction_amount: 原始提交的交易金额
        fee_amount: 手续费金额
        fee_item_type: 手续费类型（OBO_SERVICE_FEE / REMIT_SERVICE_FEE）
        net_amount: 扣除手续费后净额
        receive_amount: 换汇后实际到账金额
        account_balance: 交易后账户实时余额
        fund_move_detail: 资金流动详情
        foreign_exchange_quote: 汇率报价信息
        refund_foreign_exchange_quote: 退款汇率报价信息
        balance_type: 余额类型（NORMAL_BALANCE / SAME_NAME_TOP_UP_BALANCE / BUDGET_BALANCE）
        accounting_biz_no: 账单流水唯一 ID
        fail_reason: 失败原因（transactionStatus=FAIL 时返回）
        combined_transaction_list: 关联交易列表
        operator_info: 操作员信息
        goods_name: 商品名称
        goods_amount: 商品金额
        original_fee_amount: 优惠前手续费金额
        discount_fee_amount: 手续费优惠金额
    """

    result: Optional[Result] = None
    response_id: Optional[str] = None
    transaction_id: Optional[str] = None
    ext_transaction_id: Optional[str] = None
    transaction_status: Optional[str] = None
    transaction_time: Optional[str] = None
    transaction_type: Optional[str] = None
    transaction_amount: Optional[Amount] = None
    original_transaction_amount: Optional[Amount] = None
    fee_amount: Optional[Amount] = None
    fee_item_type: Optional[str] = None
    net_amount: Optional[Amount] = None
    receive_amount: Optional[Amount] = None
    account_balance: Optional[Amount] = None
    fund_move_detail: Optional[FundMoveDetail] = None
    foreign_exchange_quote: Optional[ForeignExchangeQuote] = None
    refund_foreign_exchange_quote: Optional[ForeignExchangeQuote] = None
    balance_type: Optional[str] = None
    accounting_biz_no: Optional[str] = None
    fail_reason: Optional[Result] = None
    combined_transaction_list: Optional[List[RelatedStatement]] = None
    operator_info: Optional[OperatorInfo] = None
    goods_name: Optional[str] = None
    goods_amount: Optional[Amount] = None
    original_fee_amount: Optional[Amount] = None
    discount_fee_amount: Optional[Amount] = None

    @classmethod
    def from_dict(cls, data: dict) -> "InquiryStatementDetailResponse":
        """从字典创建响应对象

        Args:
            data: 响应字典

        Returns:
            InquiryStatementDetailResponse 实例
        """
        if data is None:
            return cls()

        combined_list_data = data.get("combinedTransactionList")
        combined_list = None
        if combined_list_data is not None:
            combined_list = [RelatedStatement.from_dict(item) for item in combined_list_data]

        return cls(
            result=Result.from_dict(data.get("result")),
            response_id=data.get("responseId"),
            transaction_id=data.get("transactionId"),
            ext_transaction_id=data.get("extTransactionId"),
            transaction_status=data.get("transactionStatus"),
            transaction_time=data.get("transactionTime"),
            transaction_type=data.get("transactionType"),
            transaction_amount=Amount.from_dict(data.get("transactionAmount")),
            original_transaction_amount=Amount.from_dict(data.get("originalTransactionAmount")),
            fee_amount=Amount.from_dict(data.get("feeAmount")),
            fee_item_type=data.get("feeItemType"),
            net_amount=Amount.from_dict(data.get("netAmount")),
            receive_amount=Amount.from_dict(data.get("receiveAmount")),
            account_balance=Amount.from_dict(data.get("accountBalance")),
            fund_move_detail=FundMoveDetail.from_dict(data.get("fundMoveDetail")),
            foreign_exchange_quote=ForeignExchangeQuote.from_dict(data.get("foreignExchangeQuote")),
            refund_foreign_exchange_quote=ForeignExchangeQuote.from_dict(data.get("refundForeignExchangeQuote")),
            balance_type=data.get("balanceType"),
            accounting_biz_no=data.get("accountingBizNo"),
            fail_reason=Result.from_dict(data.get("failReason")),
            combined_transaction_list=combined_list,
            operator_info=OperatorInfo.from_dict(data.get("operatorInfo")),
            goods_name=data.get("goodsName"),
            goods_amount=Amount.from_dict(data.get("goodsAmount")),
            original_fee_amount=Amount.from_dict(data.get("originalFeeAmount")),
            discount_fee_amount=Amount.from_dict(data.get("discountFeeAmount")),
        )

